pkg/querier/querier.go

```diff
@@ -221,12 +204,27 @@ func (qm MergeQuerier) RemoteReadHandler(w http.ResponseWriter, r *http.Request)
 				return
 			}
 
-			matrix, err := qm.Query(ctx, from, to, matchers...)
+			iterators, err := qm.QueryRange(ctx, from, to, matchers...)
 			if err != nil {
 				errors <- err
 				return
 			}
 
+			in := metric.Interval{
+				OldestInclusive: from,
+				NewestInclusive: to,
+			}
+
+			// Convert iterators to matrix
+			matrix := make(model.Matrix, 0, len(iterators))
+			for _, it := range iterators {
+				ss := &model.SampleStream{
+					Metric: it.Metric().Metric,
+					Values: it.RangeValues(in),
```

Comment:
I've had a look into this and it doesn't look like it will panic because we are ranging over a `nil` slice which does 0 iterations. https://github.com/golang/go/blob/master/doc/go_spec.html#L5011
https://play.golang.org/p/4vvbbHMKUo